pkg/controllers/status/cluster_status_controller_test.go

```diff
@@ -0,0 +1,1121 @@
+package status
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	kubernetesfake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/sharedcli/ratelimiterflag"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/typedmanager"
+	"github.com/karmada-io/karmada/pkg/util/gclient"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+func TestClusterStatusController_Reconcile(t *testing.T) {
+	tests := []struct {
+		name            string
+		existingCluster bool
+		expectedResult  controllerruntime.Result
+		expectedError   bool
+		cluster         *clusterv1alpha1.Cluster
+	}{
+		{
+			name:            "Err is Cluster not found",
+			existingCluster: false,
+			expectedResult:  controllerruntime.Result{},
+			expectedError:   false,
+		},
+		{
+			name:            "Cluster found with finalizer",
+			existingCluster: true,
+			expectedResult:  controllerruntime.Result{},
+			expectedError:   false,
+			cluster: &clusterv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+					Finalizers: []string{
+						util.ClusterControllerFinalizer,
+					},
+				},
+			},
+		},
+		{
+			name:            "Cluster found without finalizer",
+			existingCluster: true,
+			expectedResult:  controllerruntime.Result{Requeue: true},
+			expectedError:   false,
+			cluster: &clusterv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cluster",
+					Namespace: "default",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Set up the test environment.
+			c := &ClusterStatusController{
+				Client:                 fake.NewClientBuilder().WithScheme(gclient.NewSchema()).Build(),
+				GenericInformerManager: genericmanager.GetInstance(),
+				TypedInformerManager:   typedmanager.GetInstance(),
+				ClusterClientOption: &util.ClientOption{
+					QPS:   5,
+					Burst: 10,
+				},
+				ClusterClientSetFunc: util.NewClusterClientSet,
+			}
+
+			if tt.existingCluster {
```

Comment:
We can use `tt.Cluster != nil` to replace the judgment, so we can omit the `existingCluster` parameter.